src/praisonai-agents/praisonaiagents/memory/enhanced_memory.py

```
@@ -0,0 +1,647 @@
+        if self.primary_storage:
+            # Use async storage
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+            try:
+                return loop.run_until_complete(self.search_unified(search_query))
+            finally:
+                loop.close()
+        else:
+            return self._legacy_search(search_query)
+    
+    def store_long_term(self, text: str, metadata: Dict[str, Any] = None, **kwargs):
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)